### ISSUE TITLE
Copying of a project in Minio fails.

### ISSUE THREAD

**Issue by LahariGuntha** [minio/minio-java#693]
Hi,

I am using MinioClient to copy the project in Minio. I have my code properly running in mylocalhost.But when I deploy the same code in cloud environment the operation is getting failed by throwing the below exception:

org.xmlpull.v1.XmlPullParserException: end tag name </head> must be the same as start tag <meta> from line 15 (position: TEXT seen ...</title>\n  </head>... @60:10)

  | at org.xmlpull.mxp1.MXParser.parseEndTag(MXParser.java:1699)
  | at org.xmlpull.mxp1.MXParser.nextImpl(MXParser.java:1131)
  | at org.xmlpull.mxp1.MXParser.next(MXParser.java:1093)
  | at com.google.api.client.xml.Xml.parseElementInternal(Xml.java:282)
  | at com.google.api.client.xml.Xml.parseElementInternal(Xml.java:371)
  | at com.google.api.client.xml.Xml.parseElementInternal(Xml.java:371)
  | at com.google.api.client.xml.Xml.parseElement(Xml.java:222)
  | at io.minio.messages.XmlEntity.parseXml(XmlEntity.java:75)
  | at io.minio.messages.ErrorResponse.parseXml(ErrorResponse.java:159)
  | at io.minio.messages.ErrorResponse.<init>(ErrorResponse.java:64)
  | at io.minio.MinioClient.execute(MinioClient.java:1143)
  | at io.minio.MinioClient.updateRegionCache(MinioClient.java:1221)
  | at io.minio.MinioClient.getRegion(MinioClient.java:1267)
  | at io.minio.MinioClient.executeHead(MinioClient.java:1314)
  | at io.minio.MinioClient.bucketExists(MinioClient.java:3000)

When will we be getting this kind of Issue???

**Comment by LahariGuntha** [comment]
Hi,

We have deployed Minio on Openshift using the below docker image:
docker pull icereed/openshift-minio

So the minio is present in openshift.

Earlier we used to have our code running in our local machine..and everything used to work fine.

But now we have our application containerized and launched it in Openshift. Then we started facing the Issue.

We have our java code similar to this:
 public static void main(String[] args) throws NoSuchAlgorithmException, IOException, InvalidKeyException, XmlPullParserException {
    try {
      // Create a minioClient with the Minio Server name, Port, Access key and Secret key.
      MinioClient minioClient = new MinioClient("https://play.minio.io:9000", "Q3AM3UQ867SPQQA43P2F", "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG");

      // Check if the bucket already exists.
      boolean isExist = minioClient.bucketExists("asiatrip");
      if(isExist) {
        System.out.println("Bucket already exists.");
      } else {
        // Make a new bucket called asiatrip to hold a zip file of photos.
        minioClient.makeBucket("asiatrip");
      }

      // Upload the zip file to the bucket with putObject
      minioClient.putObject("asiatrip","asiaphotos.zip", "/home/user/Photos/asiaphotos.zip");
      System.out.println("/home/user/Photos/asiaphotos.zip is successfully uploaded as asiaphotos.zip to `asiatrip` bucket.");
    } catch(MinioException e) {
      System.out.println("Error occurred: " + e);
    }

Execution is getting stopped at this point only:
 boolean isExist = minioClient.bucketExists("asiatrip");

Do we have any other way of doing this?

**Comment by balamurugana** [comment]
> Execution is getting stopped at this point only:
> boolean isExist = minioClient.bucketExists("asiatrip");

As I mentioned above the server is not sending expected XML (either error or success XML).  You could debug by having like
```java
      client = new MinioClient(endpoint, accessKey, secretKey);
      client.traceOn(System.out); // Enable trace for debugging.
```

Another option outside of java is running using `mc --debug` will help.

**Comment by nitisht** [comment]
Also `icereed/openshift-minio` is not an official Minio image @LahariGuntha Please use our official Minio Docker image from here https://hub.docker.com/r/minio/minio/

**Comment by LahariGuntha** [comment]
Hi,

Tried to deploy Minio on Openshift using Official image.
We are unable to bring up the Pod. The pod is continuously going to CrashLoopBack off state.

Here are the sequence of steps that are getting executed starting from the last while deploying the Minio in Openshift.

Back off  
Back-off restarting failed container
19 times in the last 4 minutes
	
Failed sync  
Error syncing pod
5 times in the last 4 minutes

Created 
Created container
4 times in the last 4 minutes
	
Started 
Started container
4 times in the last 4 minutes
	
Pulled 
Successfully pulled image "minio/minio"
4 times in the last 4 minutes
	
Pulling 
pulling image "minio/minio"
4 times in the last 5 minutes
	
Scheduled 
Successfully assigned minio-1-0lftg to node.test.com 

Thanks and Regards,
Lahari G

**Comment by LahariGuntha** [comment]
Hi ,


We have Minio deployed  as container in Openshift. We are trying to access Minio from another container that is deployed in same Openshift.

I am trying to create a bucket in Minio.For that first I am checking whether Bucket exists or not..
Using client.bucketExists(<BucketName>)
Here I am getting the org.xmlpull.v1.XmlPullParserException

I am again facing the same Issue again and again very frequently.

So Tried deploying Minio in Openshift using Official Docker Image provided by Minio.

But still we are facing the same Issue very frequently.

I have enabled the Debug mode.Here are the logs that I get:

---------START-HTTP---------
GET /<bucketname>/?location= HTTP/1.1
Host: <Minio-Host>:<Port No>
User-Agent: Minio (amd64; amd64) minio-java/dev
x-amz-content-sha256: hsgaduygwugdiwheoijasye6723y98udhuih782y9eudh
x-amz-date: 20181115T052RTE
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20181115/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*

HTTP/1.1 504
Content-Length: 2721
org.xmlpull.v1.XmlPullParserException: end tag name </head> must be the same as start tag <meta> from line 15 (position: TEXT seen ...</title>\n  </head>... @60:10)


I have added : System.setProperty("http.nonProxyHosts", host); Since our environment is behind proxy.
Earlier the Issue was resolved after  setting NoProxy..But now we are facing it again. 

There is no modification in the code.But it is throwing org.xmlpull.v1.XmlPullParserException very frequently. Is there any other way to resolve this Issue??

Thanks and Regards,
Lahari G

**Comment by LahariGuntha** [comment]
Hi,

Is there any other way to resolve this?

Thanks and Regards,
Lahari G

**Comment by nitisht** [comment]
@LahariGuntha are you still using `icereed/openshift-minio` ?

**Comment by LahariGuntha** [comment]
Hi Nitisht,

No we have deployed new Minio using Minio's  official docker Image.

But still we are facing the same issue again and again very frequently.

Thanks and Regards,
Lahari G

**Comment by LahariGuntha** [comment]
Hi,

Tried Using Minio client. It is working . But can we guarentee that we will not face this exception if we use mc client in a container??


But we have some Issues with it.

According to the below code:

 boolean isExist = minioClient.bucketExists("asiatrip");
  if(isExist) {
    System.out.println("Bucket already exists.");
  } else {
    minioClient.makeBucket("asiatrip");
  }

We first check if Bucket exists in Minio..

How to do this check using mc client??

Moreover we need to create projects in Minio through java.
So Installed mc clinet in container and through java we run a mc command to create a bucket and copy files recursively using below code:


String command = "/home/./mc mb play/" + <BucketName>;
String command1 = "/home/./mc cp play/" + bucket + "/ play/" + clonnedbucket + " --recursive";
Runtime rt = Runtime.getRuntime();
			Process pr = rt.exec(command);
String inputLine;
BufferedReader bfr = new BufferedReader(new InputStreamReader(pr.getInputStream()));
StringBuffer response = new StringBuffer();
while ((inputLine = bfr.readLine()) != null) {
	response.append(inputLine);
}
bfr.close();

When printing the response that we get for Minio..

If bucket is created succesfully..we are getting response as : 

Bucket created successfully `play/<BucketName>`.

Otherwise we are getting empty response. What does an empty response to refer to?

We have observed that  if bucket already exists...it is returning response as null. How can we predict what the actual issue is? How can we know that the issue is because the bucket already exists??or is it because of some other issue?

In all other cases like copying file..and deletion of buckets etc..we are getting the same issue..like getting proper message in Success scenario...but getting empty response for failure scenario..How to know what is the reason behind the failure?

How will we get the response if any exception occurs?

Thanks and Regards,
Lahari G
